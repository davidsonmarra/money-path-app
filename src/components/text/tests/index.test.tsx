import React from 'react';
import {render, screen} from '@testing-library/react-native';
import Text, {TextType} from 'src/components/text';
import getStyles from 'src/components/text/styles';

const mockText = 'Snapshot test!';
const renderComponent = (type: TextType = TextType.title) =>
  render(<Text type={type}>{mockText}</Text>);

describe('Text', () => {
  it('renders correctly', () => {
    const tree = renderComponent().toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders children correctly', () => {
    renderComponent();
    const text = screen.getByRole('text');
    expect(text.children).toEqual([mockText]);
  });

  it('renders title style correctly', () => {
    renderComponent(TextType.title);
    const text = screen.getByRole('text');
    expect(text.props.style[0]).toStrictEqual(getStyles().title);
  });

  it('renders text style correctly', () => {
    renderComponent(TextType.text);
    const text = screen.getByRole('text');
    expect(text.props.style[0]).toStrictEqual(getStyles().text);
  });

  it('renders textBold style correctly', () => {
    renderComponent(TextType.textBold);
    const text = screen.getByRole('text');
    expect(text.props.style[0]).toStrictEqual(getStyles().textBold);
  });

  it('renders buttonPrimary style correctly', () => {
    renderComponent(TextType.buttonPrimary);
    const text = screen.getByRole('text');
    expect(text.props.style[0]).toStrictEqual(getStyles().buttonPrimary);
  });

  it('renders buttonSecondary style correctly', () => {
    renderComponent(TextType.buttonSecondary);
    const text = screen.getByRole('text');
    expect(text.props.style[0]).toStrictEqual(getStyles().buttonSecondary);
  });

  it('renders header style correctly', () => {
    renderComponent(TextType.header);
    const text = screen.getByRole('text');
    expect(text.props.style[0]).toStrictEqual(getStyles().header);
  });
});
