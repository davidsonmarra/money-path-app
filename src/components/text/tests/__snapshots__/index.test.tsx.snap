// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#FFF",
        "fontSize": 24,
        "fontWeight": "bold",
      },
      undefined,
    ]
  }
>
  Snapshot test!
</Text>
`;
